src/main/java/net/minecraftforge/client/model/ModelLoader.java

```diff
@@ -375,18 +375,18 @@ public VanillaModelWrapper(ResourceLocation location, ModelBlock model, boolean
         public Collection<ResourceLocation> getTextures()
         {
             // setting parent here to make textures resolve properly
-            if(model.getParentLocation() != null)
+            if(model.getParentLocation() != null && model.parent == null)
             {
                 if(model.getParentLocation().getResourcePath().equals("builtin/generated"))
                 {
                     model.parent = MODEL_GENERATED;
                 }
                 else
                 {
-                    IModel parent = ModelLoaderRegistry.getModelOrLogError(model.getParentLocation(), "Could not load vanilla model parent '" + model.getParentLocation() + "' for '" + model);
-                    if(parent instanceof VanillaModelWrapper)
+                    Optional<ModelBlock> vanillaParent = ModelLoaderRegistry.getModelOrLogError(model.getParentLocation(), "Could not load vanilla model parent '" + model.getParentLocation() + "' for '" + model).asVanillaModel();
```

Comment:
Better expressed as:
```suggestion
                    model.parent = ModelLoaderRegistry.getModelOrLogError(model.getParentLocation(), "Could not load vanilla model parent '" + model.getParentLocation() + "' for '" + model).asVanillaModel()
                        .orElseThrow(() -> new IllegalStateException("vanilla model '" + model + "' can't have non-vanilla parent");
```